samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -579,16 +667,12 @@ static struct bt_mesh_sensor_srv chip_temp_sensor_srv =
 
 static struct k_work_delayable presence_detected_work;
 
+#define BOOLEAN_INIT(_bool) { .format = &bt_mesh_sensor_format_boolean, .raw = { !!(_bool) } }
```

Comment:
Yes. A C boolean can contain any number. The `raw` field can only legally contain 1 or 0. So it's there because the format spec is stricter than C is.

Reply:
> A C boolean can contain any number.

This I didn't get. Could you point me to the C99 standard where it would say this? You use `BOOLEAN_INIT` with `true` and `false` which are either 1 or 0, not any other number by standard. If you were thinking of using this macro with any number then, probably, fine (though the argument name is confusing), but from what I see how the macro is used in the sample `!!` is not needed.

Reply:
Right, I was confused, I thought the C99 boolean type still did the old thing where it just was an `int`. But you are correct, it will only ever be 1 or 0. I'll remove the `!!`.

Reply:
This is what I found (though I'm not an expert in the standard and may treat it incorrectly):
> 6.3.1.2 Boolean type
When any scalar value is converted to _Bool, the result is 0 if the value compares equal to 0; otherwise, the result is 1.